### PR TITLE
Routing table persistence

### PR DESCRIPTION
**Pull request by cloudhead**

### CODE REVIEW

File: radicle-node/src/client.rs [hunk 1]
```diff
@@ -73,24 +89,27 @@ impl<R: Reactor> Client<R> {
...
 
-    pub fn run(mut self, config: Config) -> Result<(), nakamoto_net::error::Error> {
+    pub fn run(mut self, config: Config) -> Result<(), Error> {
         let network = config.service.network;
         let rng = fastrand::Rng::new();
         let time = LocalTime::now();
         let storage = self.profile.storage;
         let signer = self.profile.signer;
         let addresses = HashMap::with_hasher(rng.clone().into());
+        let routing = routing::Table::open("routing.db")?;
```
Comment (slack-coder) [inline radicle-node/src/client.rs:99]: Would making it a configuration based path make it better?  Like path::join(config.data, "routing.db")

File: radicle-node/src/service.rs [hunk 2]
```diff
@@ -124,15 +122,15 @@ pub enum Command {
...
+pub struct Service<R, A, S, G> {
     /// Service configuration.
     config: Config,
     /// Our cryptographic signer and key.
     signer: G,
     /// Project storage.
     storage: S,
-    /// Tracks the location of projects.
-    routing: Routing,
+    /// Network routing table. Keeps track of where projects are located.
```
Comment (slack-coder) [inline radicle-node/src/service.rs:132]: nitpick, the first period may not be required

File: radicle-node/src/service.rs [hunk 3]
```diff
@@ -612,7 +609,8 @@ where
                 } else {
                     return Ok(false);
                 }
-                self.process_inventory(&message.inventory, *node, git);
+                self.process_inventory(&message.inventory, *node, git)
+                    .unwrap();
```
Comment (slack-coder) [inline radicle-node/src/service.rs:613]: Is this known to never panic?  Maybe a comment why if so would help

**Comment by slack-coder** [comment]
otherwise looks good to me

File: radicle-node/src/client.rs (hunk 1 above)
Comment (cloudhead) [reply radicle-node/src/client.rs:99]: Ah yes, forgot to do this, good catch!

File: radicle-node/src/service.rs (hunk 3 above)
Comment (cloudhead) [reply radicle-node/src/service.rs:613]: Nah, we're not there yet, ie. there are *lots* of unwraps in the codebase that need to be handled correctly. We'll have to do a pass at some point and fix them.

File: radicle-node/src/service/routing/schema.sql [hunk 4]
```diff
@@ -0,0 +1,15 @@
+--
+-- Routing table SQL schema.
+--
+create table if not exists "routing" (
+  "id"           integer   primary key,
```
Comment (adaszko) [inline radicle-node/src/service/routing/schema.sql:5]: It’s a matter of style of course but the quotes shouldn’t be necessary

File: radicle-node/src/service.rs [hunk 5]
```diff
@@ -124,15 +122,15 @@ pub enum Command {
...
     /// Service configuration.
     config: Config,
     /// Our cryptographic signer and key.
     signer: G,
     /// Project storage.
     storage: S,
-    /// Tracks the location of projects.
-    routing: Routing,
+    /// Network routing table. Keeps track of where projects are located.
+    routing: R,
```
Comment (adaszko) [inline radicle-node/src/service.rs:133]: I’d use a concrete type here for simplicity but you may have more context on what’s going to happen with this portion of code in the near term

File: radicle/Cargo.toml [hunk 6]
```diff
@@ -8,6 +8,7 @@ edition = "2021"
 [features]
 default = []
 test = ["quickcheck"]
+sql = ["rusqlite"]
```
Comment (adaszko) [inline radicle/Cargo.toml:11]: Nitpicking: “sql” sounds a bit imprecise to me.  How about “persistent_routing_table”?

**Review by adaszko — APPROVED** [review]

File: radicle/Cargo.toml (hunk 6 above)
Comment (cloudhead) [reply radicle/Cargo.toml:11]: The crate feature is just enabling the `From/ToSql` instances

**Pull request merged**